Scripts/Core/EditorVR.Rays.cs

```diff
@@ -441,22 +458,51 @@ internal void PreProcessRaycastSource(Transform rayOrigin)
 
 			static Node? RequestNodeFromRayOrigin(Transform rayOrigin)
 			{
-				Node? node = null;
-				var evrDeviceData = evr.m_DeviceData;
-				for (var i = 0; i < evrDeviceData.Count; i++)
+				foreach (var deviceData in evr.m_DeviceData)
 				{
-					var deviceData = evrDeviceData[i];
 					if (!deviceData.proxy.active)
 						continue;
 
 					if (deviceData.rayOrigin == rayOrigin)
+						return deviceData.node;
+				}
+
+				foreach (var kvp in evr.GetNestedModule<MiniWorlds>().rays)
+				{
+					if (kvp.Key == rayOrigin)
+						return kvp.Value.node;
+				}
+
+				return null;
+			}
+
+			static void SetDefaultRayColor(Transform rayOrigin, Color color)
+			{
+				if (rayOrigin)
+				{
+					var dpr = rayOrigin.GetComponentInChildren<DefaultProxyRay>();
+					if (dpr)
+					{
+						dpr.SetColor(color);
+					}
+				}
+
+				var highlightModule = evr.GetModule<HighlightModule>();
+				highlightModule.highlightColor = color;
+			}
+
+			static Color GetDefaultRayColor(Transform rayOrigin)
+			{
+				if (rayOrigin)
+				{
+					var dpr = rayOrigin.GetComponentInChildren<DefaultProxyRay>();
+					if (dpr)
 					{
-						node = deviceData.node;
-						break;
+						return dpr.GetColor();
 					}
 				}
 
-				return node;
+				return default(Color);
```

Comment:
Hm. I think I do this because this gets called once or twice before highlightModule is instantiated. I'll check it out.